qcommon/src/main/kotlin/jake2/qcommon/EntityParser.kt

```diff
@@ -0,0 +1,76 @@
+package jake2.qcommon
+
+/**
+ * Parse the entities string.
+ * Very similar to the json format, but without a colon between key/value pairs
+ * and without commas between objects, like
+ *
+ *    {
+ *     "key" "value"
+ *    }
+ *    {
+ *     "foo" "bar"
+ *    }
+ *
+ * TODO: add some validation and error messages, because right now parser just silently skips anything it doesn't like
```

Comment:
I think using some lightweight parser library will give you this for free

Reply:
this one? 
https://github.com/h0tk3y/better-parse

looks interesting